payment.yaml

```diff
@@ -0,0 +1,227 @@
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: Payment Ecosystem
+  description: Internal API that provides payment and wallet creation services
+servers:
+  - url: 'https://api.kinmarketplace.internal/v1/'
+paths:
+  /orders/{order_id}:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+      - in: path
+        name: order_id
+        schema:
+          type: string
+        required: true
+        description: application order id
+    get:
+      tags:
+        - Orders
+      description: get order info
+      summary: get order info
+      operationId: getOrder
+      responses:
+        '200':
+          description: Successfully returned order info
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Order'
+        '404':
+          description: Order not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /orders:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+    post:
+      tags:
+        - Orders
+      description: create an payment order
+      summary: create an payment order
+      operationId: createOrder
+      requestBody:
+        description: payment description
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Payment'
+      responses:
+        '201':
+          description: Successfully created an order and now is processing
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+   
+  /wallets:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+    post:
+      tags:
+        - Wallets
+      description: create wallet
+      summary: create wallet
+      operationId: createWallet
+      requestBody:
+        description: payment description
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/WalletAddress'
+
+      responses:
+        '201':
+          description: Successfully created wallet
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
```

Comment:
General Error missing

Reply:
if wallet exists I return 202 as in an idempotent request. The request is async